cpp/ql/lib/semmle/code/cpp/exprs/Expr.qll

```diff
@@ -58,6 +58,21 @@ class Expr extends StmtParent, @expr {
   /** Gets the parent of this expression, if any. */
   Element getParent() { exprparents(underlyingElement(this), _, unresolveElement(result)) }
 
+  /**
+   * Gets the `n`th compiler-generated destructor call that is performed after this expression, in
+   * order of destruction.
+   */
+  DestructorCall getSyntheticDestructor(int n) {
```

Comment:
Could we avoid using the term "synthetic" in the public API? I don't actually know why the relational is using this term, because I don't think there's anything more "synthetic" about these destructor calls than other things in the AST

Reply:
Maybe `getImplicitDestructor` instead?

Reply:
Sounds good to me. Maybe even `getImplicitDestructorCall` since this is really the _call_ to the destructor?